pkg/sink/sink_test.go

```diff
@@ -179,687 +179,758 @@ func checkSinkResponse(t *testing.T, resp *http.Response, elName string) {
-
-	resources := test.Resources{
-		TriggerBindings:  tbs,
-		TriggerTemplates: []*triggersv1.TriggerTemplate{tt},
-		EventListeners:   []*triggersv1.EventListener{el},
-		Triggers:         triggers,
+		})
 	}
 
-	sink, dynamicClient := getSinkAssets(t, resources, el.Name, DefaultAuthOverride{})
-
-	ts := httptest.NewServer(http.HandlerFunc(sink.HandleEvent))
-	defer ts.Close()
-
-	resp, err := http.Post(ts.URL, "application/json", bytes.NewReader(eventBody))
-	if err != nil {
-		t.Fatalf("Error creating Post request: %s", err)
+	tenGitCloneTaskRuns := []pipelinev1.TaskRun{}
+	for i := 0; i < 10; i++ {
+		tr := gitCloneTaskRun.DeepCopy()
+		tr.Name = fmt.Sprintf("git-clone-run-%d", i)
+		tr.Labels["triggers.tekton.dev/trigger"] = fmt.Sprintf("git-clone-trigger-%d", i)
+		tenGitCloneTaskRuns = append(tenGitCloneTaskRuns, *tr)
 	}
 
-	checkSinkResponse(t, resp, el.Name)
-	// Check right resources were created.
-	var wantPrs []pipelinev1alpha1.PipelineResource
-	wantResource := pipelinev1alpha1.PipelineResource{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "tekton.dev/v1alpha1",
-			Kind:       "PipelineResource",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "my-pipelineresource",
-			Namespace: triggerNS,
-			Labels: map[string]string{
-				"app":         "bar\t\r\nbaz昨",
-				"type":        "application/json",
-				resourceLabel: "my-eventlistener",
-				triggerLabel:  "my-trigger",
-				eventIDLabel:  eventID,
-			},
-		},
-		Spec: pipelinev1alpha1.PipelineResourceSpec{
-			Type: pipelinev1alpha1.PipelineResourceTypeGit,
-			Params: []pipelinev1alpha1.ResourceParam{
-				{Name: "url", Value: "testurl"},
-				{Name: "revision", Value: "testrevision"},
-			},
-		},
-	}
-	wantPrs = append(wantPrs, wantResource)
-	// Sort actions (we do not know what order they executed in)
-	gotPrs := getCreatedPipelineResources(t, dynamicClient.Actions())
-	if diff := cmp.Diff(wantPrs, gotPrs, cmpopts.SortSlices(comparePR)); diff != "" {
-		t.Errorf("Created resources mismatch (-want + got): %s", diff)
-	}
-}
-
-func TestHandleEventTriggerRef(t *testing.T) {
-	eventBody := json.RawMessage(`{"head_commit": {"id": "testrevision"}, "repository": {"url": "testurl"}, "foo": "bar\t\r\nbaz昨"}`)
-
-	pipelineResource := pipelinev1alpha1.PipelineResource{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "tekton.dev/v1alpha1",
-			Kind:       "PipelineResource",
+	tests := []struct {
+		name string
+		// resources are the K8s objects to setup the test env.
+		resources test.Resources
+		// webhookInterceptor is a handler that implements any webhook interceptors referenced in the triggers.
+		webhookInterceptor http.HandlerFunc
+		eventBody          []byte
+		headers            map[string][]string
+		// want is the resulting TaskRun(s) created by the EventListener
+		want []pipelinev1.TaskRun
+	}{{
+		name: "single trigger embedded within EventListener",
```

Comment:
Are there any conventions on test case naming? I generally like this style of naming but have noticed that it is inconsistent throughout the `triggers` project.

Reply:
no hard and fast rule. Ideally the name would be both descriptive and short but if not possible, I prefer a more descriptive but longer name over a shorter one (though you are right we haven't been very consistent about this).